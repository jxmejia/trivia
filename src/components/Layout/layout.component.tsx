import { memo } from "react";
import { LayoutProps } from "./layout.definition";
import { StyledLayout } from "./layout.style";

const Layout = ({ children }: LayoutProps): JSX.Element => {
  return <StyledLayout>{children}</StyledLayout>;
};

export default memo(Layout);
