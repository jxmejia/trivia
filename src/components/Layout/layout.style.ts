import styled from "@emotion/styled";

export const StyledLayout = styled.div`
  text-align: center;
`;
