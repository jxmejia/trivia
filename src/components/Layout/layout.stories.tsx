import { ComponentStory, ComponentMeta } from "@storybook/react";
import Layout from "./layout.component";

export default {
  title: "Layout",
  component: Layout,
} as ComponentMeta<typeof Layout>;

const Template: ComponentStory<typeof Layout> = (args) => <Layout {...args} />;

export const Default = Template.bind({});

Default.args = {
  children: "Trivia challenge",
};
