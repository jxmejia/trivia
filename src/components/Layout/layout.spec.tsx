import { createSerializer } from "@emotion/jest";
import { render } from "@testing-library/react";
import Layout from "./layout.component";

expect.addSnapshotSerializer(createSerializer());

it("renders component", () => {
  const { container } = render(<Layout />);
  expect(container).toMatchSnapshot();
});
