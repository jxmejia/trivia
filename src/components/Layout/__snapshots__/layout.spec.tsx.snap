// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout is renders component 1`] = `
.emotion-0 {
  text-align: center;
}

<div>
  <div
    class="emotion-0"
  />
</div>
`;

exports[`Layout renders component 1`] = `
.emotion-0 {
  text-align: center;
}

<div>
  <div
    class="emotion-0"
  />
</div>
`;

exports[`renders component 1`] = `
.emotion-0 {
  text-align: center;
}

<div>
  <div
    class="emotion-0"
  />
</div>
`;
